dart/lib/vm_service_lib.dart

```diff
@@ -1427,18 +1476,41 @@ class Class extends Obj {
 
   Class._fromJson(Map<String, dynamic> json) : super._fromJson(json) {
     name = json['name'];
-    error = _createObject(json['error']);
+    error = createServiceObject(json['error']);
     isAbstract = json['abstract'];
     isConst = json['const'];
-    library = _createObject(json['library']);
-    location = _createObject(json['location']);
-    superClass = _createObject(json['super']);
-    superType = _createObject(json['superType']);
-    interfaces = new List<InstanceRef>.from(_createObject(json['interfaces']));
-    mixin = _createObject(json['mixin']);
-    fields = new List<FieldRef>.from(_createObject(json['fields']));
-    functions = new List<FuncRef>.from(_createObject(json['functions']));
-    subclasses = new List<ClassRef>.from(_createObject(json['subclasses']));
+    library = createServiceObject(json['library']);
+    location = createServiceObject(json['location']);
+    superClass = createServiceObject(json['super']);
+    superType = createServiceObject(json['superType']);
+    interfaces =
+        new List<InstanceRef>.from(createServiceObject(json['interfaces']));
+    mixin = createServiceObject(json['mixin']);
+    fields = new List<FieldRef>.from(createServiceObject(json['fields']));
+    functions = new List<FuncRef>.from(createServiceObject(json['functions']));
+    subclasses =
+        new List<ClassRef>.from(createServiceObject(json['subclasses']));
+  }
+
+  Map<String, dynamic> toJson() {
+    var json = super.toJson();
+    json['type'] = 'Class';
+    json.addAll({
+      'name': name,
+      'abstract': isAbstract,
+      'const': isConst,
+      'library': library.toJson(),
+      'interfaces': interfaces.map((f) => f.toJson()).toList(),
+      'fields': fields.map((f) => f.toJson()).toList(),
+      'functions': functions.map((f) => f.toJson()).toList(),
+      'subclasses': subclasses.map((f) => f.toJson()).toList(),
+    });
+    _setIfNotNull(json, 'error', error?.toJson());
```

Comment:
These look good. I suppose the method calls will be in-lined (in dart2js and the vm)?

Another - equally valid - way would be to inline manually (similar to what you had before):

```dart
if (location != null) {
  json['location'] = location.toJson();
}
```

Reply:
Ya, I would hope it gets inlined, but we can always follow up if performance becomes an issue. Generally, these payloads don't tend to be huge from what I have seen anyways.